IOPool/Streamer/src/StreamerOutputModuleCommon.cc

```diff
@@ -237,8 +237,12 @@ namespace edm {
       double timeInSec =
           static_cast<double>(now.tv_sec) + (static_cast<double>(now.tv_usec) / 1000000.0) - timeInSecSinceUTC;
       // what about overflows?
-      if (lumiSectionInterval_ > 0)
+      if (lumiSectionInterval_ > 0) {
         lumi = static_cast<uint32>(timeInSec / lumiSectionInterval_) + 1;
+      } else {
+        assert(not(lumiSectionInterval_ < 0));
+        lumi = e.luminosityBlock();
```

Comment:
I'm a bit confused. With this change negative `lumiSectionInterval_` values will lead to assertion failure. Would it be better to catch that in the `StreamerOutputModuleCommon` constructor and throw an exception in that case?

Or, if negative `lumiSectionInterval_` would be ok, how about changing the code structure here along
```cpp
if (lumiSectionInterval_ <= 0) {
  lumi = e.luminosityBlock();
} else {
   ...
}
```
?

Reply:
I can't see how a negative section interval can possibly be OK. I only gave a value after the assert as that was necessary to stop the compiler warning.

Reply:
How about then
* check the validity of the configuration parameter already in `StreamerOutputModuleCommon::Parameters StreamerOutputModuleCommon::parameters()` and throw an exception if it is negative
  * In principle it would be nice to update the `fillDescriptions()` to `"If greater than 0"
* make the `lumiSectionInterval_` an `int const`
* If the compiler doesn't figure out that negative values are not possible (I could believe that), then use `if (lumiSectionInterval_ <= 0)`?